src/ConnectAuth.ts

```diff
@@ -0,0 +1,192 @@
+import { AuthError, AuthErrorCode } from './AuthError';
+import { decodeUnverifiedConnectJwt, verifyConnectJwt } from './Jwt';
+import { RequestReader } from './requestReader';
+import { ConnectJwt, ContextQsh, CredentialsLoader, InstallType } from './types';
+
+interface CommonVerifyArgs<E> {
+  baseUrl: string;
+  requestReader: RequestReader;
+  loadCredentials: CredentialsLoader<E>;
+  skipQueryStringHashCheck?: boolean; // Bitbucket does not implement Query String Hash
```

Comment:
Instead of passing a `product` and having implicitly different code paths for Bitbucket, I think it makes more sense to explicitly pass an argument that makes it clear what happens underneath.
In this case, Connie and Jira apps will not need to pass this but Bitbucket apps will always pass true here if my understanding is correct.

Reply:
I had a wrong assumption based on the previous implementation that skipped QSH checking when product !== bb. Will fix that.